### PR TITLE
TIP-1250: upgrade Elasticsearch to 7.3

### PR DESCRIPTION
**Pull request by jjanvier**
This PR aims to make the PIM work with Elasticsearch 7.3. We'll probably upgrade to latest ES7.x before the next major release.

ES7.0 comes with a few [BC Breaks](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/breaking-changes-7.0.html).

Here are the changes that impact us the most:
- [mapping types have been removed](https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html)
- [_uid has been removed](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/mapping-uid-field.html)
- [total hits is now approximative](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/search-request-body.html#request-body-search-track-total-hits) 
- [split_on_whitespace search option has been removed](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/breaking-changes-7.0.html)
- [minimum_should_match should be explicit now](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#_the_filter_context_has_been_removed)

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/CategoryTree/ListRootCategoriesWithCountNotIncludingSubCategories.php [hunk 1]
```diff
@@ -114,7 +109,7 @@ private function countProductInCategories(array $categoriesWithoutCount, int $ro
             ];
         }
 
-        $rows = $this->client->msearch($this->indexType, $body);
+        $rows = $this->client->msearch($body);
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/CategoryTree/ListRootCategoriesWithCountNotIncludingSubCategories.php]: you should set `track_total_hits` I think ( same for all queries, and in EE also)

Comment (jjanvier) [reply src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/CategoryTree/ListRootCategoriesWithCountNotIncludingSubCategories.php]: @ahocquard I tried to do that everywhere, but I certainly forgot some
it's difficult to know all the places where it should be added

File: docker-compose.yml [hunk 2]
```diff
@@ -82,7 +82,7 @@ services:
       - 'pim'
 
   elasticsearch:
-    image: 'docker.elastic.co/elasticsearch/elasticsearch-oss:6.5.4'
+    image: 'docker.elastic.co/elasticsearch/elasticsearch:7.3.1'
```
Comment (momoss) [inline docker-compose.yml]: We don't use the `-oss` elasticsearch  version anymore?

Comment (momoss) [reply docker-compose.yml]: And by the way the last version is 7.3.2

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/CategoryTree/ListRootCategoriesWithCountNotIncludingSubCategories.php (hunk 1 above)
Comment (ahocquard) [reply src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/CategoryTree/ListRootCategoriesWithCountNotIncludingSubCategories.php]: This is counting data in the category tree, I can help you if you need

File: docker-compose.yml (hunk 2 above)
Comment (jjanvier) [reply docker-compose.yml]: @momoss I didn't find this repository :man_shrugging:

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/CategoryTree/ListRootCategoriesWithCountNotIncludingSubCategories.php (hunk 1 above)
Comment (jjanvier) [reply src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/CategoryTree/ListRootCategoriesWithCountNotIncludingSubCategories.php]: fixed in those classes and in EE too

File: docker-compose.yml (hunk 2 above)
Comment (momoss) [reply docker-compose.yml]: You can find it here https://www.elastic.co/downloads/past-releases/elasticsearch-oss-7-3-2 ;)

Comment (jjanvier) [reply docker-compose.yml]: ah ok and what is the difference?

Comment (momoss) [reply docker-compose.yml]: The non oss contains x-pack who is the security layer of elasticsearch , you can find more explanations [here](https://www.elastic.co/what-is/open-x-pack)

Comment (jjanvier) [reply docker-compose.yml]: Thanks momoss! I fixed in another PR

**Pull request merged**